src/JMS/Serializer/JsonSerializationVisitor.php

```diff
@@ -158,7 +158,9 @@ public function visitProperty(PropertyMetadata $metadata, $data, Context $contex
         $v = $this->accessor->getValue($data, $metadata);
 
         $v = $this->navigator->accept($v, $metadata->type, $context);
-        if (null === $v && $context->shouldSerializeNull() !== true) {
+        if ((null === $v && $context->shouldSerializeNull() !== true)
+            || ($v instanceof \ArrayObject && 0 === count($v) && $metadata->skipWhenEmpty)
```

Comment:
So a plain empty array is not skipped as it checks for \ArrayObject? That's what I hoped for with #124